src/actions/Deselect.ts

```diff
@@ -0,0 +1,41 @@
+import { Selection } from "vscode";
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  TypedSelection,
+  Graph,
+} from "../typings/Types";
+import { runOnTargetsForEachEditor } from "../util/targetUtils";
+
+export default class Deselect implements Action {
+  getTargetPreferences: () => ActionPreferences[] = () => [
+    { insideOutsideType: "inside" },
+  ];
+
+  constructor(private graph: Graph) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    await runOnTargetsForEachEditor(targets, async (editor, targets) => {
+      // Remove selections with a non-empty intersection
+      const newSelections = editor.selections.filter(
+        (selection) =>
+          !targets.find((target) => {
+            const intersection =
+              target.selection.selection.intersection(selection);
+            return intersection && !intersection.isEmpty;
+          })
+      );
+      // The editor requires at least one selection. Keep "primary" selection active
+      editor.selections = newSelections.length
+        ? newSelections
+        : [new Selection(editor.selection.active, editor.selection.active)];
+    });
+
+    return {
+      thatMark: targets.map((target) => target.selection),
```

Comment:
Do you think the `that` mark would be more useful if it were the selections that got dropped? We could probably just leave it for now and see how it works in practice but just a thought

Reply:
I think we can leave it like this and try it out. In most cases that is token you actually referenced. But I do agree that could potentially be useful. 

Maybe we can use the source mark?

Reply:
Hmm not sure source mark is right here but interesting idea.  Agreed tho let's just see if we ever actually want a way to refer to the removed selections in practice